src/main/java/mate/academy/dao/BookDaoImplementation.java

```diff
@@ -0,0 +1,104 @@
+package mate.academy.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.util.ConnectionUtil;
+
+@Dao
+public class BookDaoImplementation implements BookDao {
+    @Override
+    public Book create(Book book) {
+        String query = "INSERT INTO books (title, price) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query,
+                        PreparedStatement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.executeUpdate();
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                book.setId(generatedKeys.getLong(1));
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't insert book into the database: " + book, e);
```

Comment:
Consider using a custom exception class, such as `DataProcessingException`, instead of `RuntimeException` for better error handling and clarity. This will also align with the exception handling used in the `Main.java` file.